src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -91,64 +94,101 @@ public function load($data, $type = null): RouteCollection
 
-            if (null === $resourceShortName) {
-                throw new InvalidResourceException(sprintf('Resource %s has no short name defined.', $resourceClass));
-            }
+            foreach ($resourceMetadataCollection as $resourceMetadata) {
+                $resourceShortName = $resourceMetadata->getShortName();
 
-            if (null !== $collectionOperations = $resourceMetadata->getCollectionOperations()) {
-                foreach ($collectionOperations as $operationName => $operation) {
-                    $this->addRoute($routeCollection, $resourceClass, $operationName, $operation, $resourceMetadata, OperationType::COLLECTION);
+                if (null === $resourceShortName) {
+                    throw new InvalidResourceException(sprintf('Resource %s has no short name defined.', $resourceClass));
                 }
-            }
 
-            if (null !== $itemOperations = $resourceMetadata->getItemOperations()) {
-                foreach ($itemOperations as $operationName => $operation) {
-                    $this->addRoute($routeCollection, $resourceClass, $operationName, $operation, $resourceMetadata, OperationType::ITEM);
+                if ($resourceMetadata->isNewResource()) {
+                    foreach ($resourceMetadata->getOperations() as $operationName => $operation) {
+                        try {
+                            $identifiers = $operation->identifiers ?? $resourceMetadata->getAttribute('identifiers', $this->identifiersExtractor ? $this->identifiersExtractor->getIdentifiersFromResourceClass($resourceClass) : ['id']);
+                        // TODO: maybe don't throw an Exception when we look into Collections they may not have any
+                        } catch (RuntimeException $e) {
+                            $identifiers = [];
+                        }
+
+                        $route = new Route(
+                            ($operation->routePrefix ?? '') . $operation->uriTemplate,
+                            [
+                                '_controller' => $operation->controller,
+                                '_format' => null,
+                                '_stateless' => $operation->stateless,
+                                '_api_resource_class' => $resourceClass,
+                                '_api_identifiers' => $identifiers,
+                                '_api_has_composite_identifier' => $operation->compositeIdentifier,
+                                '_api_operation_name' => $operationName,
+                            ] + ($operation->defaults ?? []),
+                            $operation->requirements ?? [],
+                            $operation->options ?? [],
+                            $operation->host ?? '',
+                            $operation->schemes ?? [],
+                            [$operation->method],
+                            $operation->condition ?? ''
```

Comment:
Should we implement `ArrayObject` or `ArrayAccess` on `Operation` @dunglas ? This would help with OpenApi/Swagger and related to make the transition smoother.